### ISSUE TITLE
Import: Low-res image used as primary file in a stack

### ISSUE THREAD

**Issue by uxusoft** [photoprism/photoprism#813]
Hello,

On iPhone it is possible to take "Live" photos. Unfortunately, when I import such live photos in photoprism, it is importing as movie with wrong low-resolution screenshot as main picture.

For example, this is "Live" photo imported directly from iPhone (no EXIF data because iPhone deleted it automatically):
<img width="289" alt="Screenshot 2021-01-04 at 18 52 44" src="https://user-images.githubusercontent.com/64733669/103564002-1059b780-4ebe-11eb-9df1-b5efa8204b59.png">

And this is the same photo, but imported from Mac using same picture, but as HEIC+MOV+XMP (exported from Photos app):

<img width="333" alt="Screenshot 2021-01-04 at 18 53 52" src="https://user-images.githubusercontent.com/64733669/103564082-35e6c100-4ebe-11eb-8a6e-a7532039e125.png">

<img width="245" alt="Screenshot 2021-01-04 at 18 54 19" src="https://user-images.githubusercontent.com/64733669/103564101-45660a00-4ebe-11eb-8aec-9471c55054fa.png">

Picture is now "Live" but with wrong low-resolution main photo.

```
time="2021-01-04T17:48:48Z" level=info msg="moving media files from /photoprism/import to /photoprism/originals"
time="2021-01-04T17:48:48Z" level=info msg="classify: loading nasnet"
time="2021-01-04T17:48:51Z" level=info msg="classify: loading labels from labels.txt"
time="2021-01-04T17:48:51Z" level=info msg="import: no .ppignore file found"
time="2021-01-04T17:48:51Z" level=info msg="media: IMG_7098.mov was taken at 2021-01-04 17:48:22.977787757 +0000 UTC (file mod time)"
time="2021-01-04T17:48:51Z" level=info msg="import: moving main mov file upload/IMG_7098.mov to 2021/01/20210104_174822_0E0DB223.mov"
time="2021-01-04T17:48:51Z" level=info msg="import: moving related xmp file upload/IMG_7098.xmp to 2021/01/20210104_174822_0E0DB223.xmp"
time="2021-01-04T17:48:51Z" level=info msg="import: moving related heif file upload/IMG_7098.HEIC to 2021/01/20210104_174822_0E0DB223.heic"
time="2021-01-04T17:48:52Z" level=info msg="media: 13 thumbnails created for 20210104_174822_0E0DB223 [303.697123ms]"
time="2021-01-04T17:48:52Z" level=info msg="import: added main mov file 2021/01/20210104_174822_0E0DB223.mov"
time="2021-01-04T17:48:52Z" level=info msg="import: added related xmp file 2021/01/20210104_174822_0E0DB223.xmp"
time="2021-01-04T17:48:52Z" level=info msg="import: added related heif file 2021/01/20210104_174822_0E0DB223.heic"
time="2021-01-04T17:48:54Z" level=info msg="import: added related jpg file /photoprism/storage/sidecar/2021/01/20210104_174822_0E0DB223.mov.jpg"
time="2021-01-04T17:48:54Z" level=info msg="import: deleted empty folder upload"
time="2021-01-04T17:48:54Z" level=info msg="import completed in 6.309828303s"
time="2021-01-04T17:48:54Z" level=info msg="closed database connection"
```

Test photo:
[HEIC](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_7098.HEIC)
[MOV](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_7098.mov)
[XMP](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_7098.xmp)

**Comment by uxusoft** [comment]
A huge amount of Live photos are already imported. It seems that it is needed to make HEIC photo as main photo, not the MOV.jpg screenshot.

**Comment by lastzero** [comment]
Depending on the import / indexing order, the "wrong" primary (preview) image may be selected for a movie or raw. Right now, it simply is the first file.

**Comment by uxusoft** [comment]
Thank you for reply. It seems that all heif photos (thousands) with low-res mov.jpg thumbnail. And more important, when I click on picture- I see this thumbnail, there is no way to see/download original high-res picture.

Maybe it is possible to fix it somehow by deleting all mov.jpg and reindexing? Could you please advise any method?

**Comment by lastzero** [comment]
Can you send a zip with examples for testing? That's better than guessing :)

**Comment by uxusoft** [comment]
> Test photo:
> [HEIC](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_7098.HEIC)
> [MOV](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_7098.mov)
> [XMP](https://test-zxb.s3.eu-central-1.amazonaws.com/IMG_7098.xmp)

[ZIP](https://test-zxb.s3.eu-central-1.amazonaws.com/example_heic_live.zip)

**Comment by lastzero** [comment]
Thanks! I'll take a look at this later this week, or maybe @graciousgrey can test it... I'm blocked until Thursday.

**Comment by graciousgrey** [comment]
I can confirm this. 
@uxusoft you can download the .heic file from the files tab within the edit dialogue.

We will fix this, the jpg should be created from the heic in such cases

**Comment by uxusoft** [comment]
Wow! Thank you very much for quick fix!
I am going to check it, but could you please also advise the correct way to update Photoprism?
I assume that I have also to delete all .*.mov.jpg.

Usually I do this way, but not sure that it is correct (and fastest):
1. docker-compose down
2. docker system prune -a
3. rm -rf .photoprism
4. docker-compose up -d
5. docker exec photoprism_photoprism_1 photoprism index -a

**Comment by lastzero** [comment]
New preview builds available for testing, see https://docs.photoprism.org/getting-started/updates/

TL;DR: Development previews are tagged with `photoprism/photoprism:preview` instead of `photoprism/photoprism:latest`. Update your config to use this for testing. We personally also deploy the previews on production, but that choice depends on your stability requirements.

**Comment by lastzero** [comment]
Even with this fix, you MAY need to manually change the primary image to the one you want to use if there are multiple JPEGs due to multiple non-JPEG media files in a stack. Note that indexing happens concurrently, so a final decision can only be made after indexing is complete. We may later auto-select the best version in our background metadata worker.

**Comment by uxusoft** [comment]
@graciousgrey @lastzero Thank you very much!
I will try to test this fix. Thank you for pointing to "preview" tag. I will change it, delete whole .photoprism folder and index.

**Comment by uxusoft** [comment]
Only several pictures was fixed. I decided to re-import everything. First added to archive incorrect pictures. But during import get errors that images already exists, for example:
import: IMG_7079.HEIC is identical to 2021/01/20210104_173456_6A009A09.heic

But there are no option to delete images. So I have again to start over with fresh install and collection of tens of thousand photos. It is very hard to debug without delete option.
